reco_utils/recommender/vowpal_wabbit/vw.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+This file provides a wrapper to run Vowpal Wabbit from the command line through python.
+It is not recommended to use this approach in production, there are python bindings that can be installed from the
+repository or pip or the command line can be used. This is merely to demonstrate vw usage in the example notebooks.
+"""
+
+import os
+from subprocess import run
+from tempfile import TemporaryDirectory
+
+import pandas as pd
+
+
+from reco_utils.common.constants import (
+    DEFAULT_USER_COL,
+    DEFAULT_ITEM_COL,
+    DEFAULT_RATING_COL,
+    DEFAULT_TIMESTAMP_COL,
+    PREDICTION_COL,
+)
+
+
+class VW:
+    """Vowpal Wabbit Class"""
+
+    def __init__(
+        self,
+        col_user=DEFAULT_USER_COL,
+        col_item=DEFAULT_ITEM_COL,
+        col_rating=DEFAULT_RATING_COL,
+        col_timestamp=DEFAULT_TIMESTAMP_COL,
+        col_prediction=PREDICTION_COL,
+        **kwargs,
+    ):
+        """Initialize model parameters
+        Args:
+        """
+
+        # create temporary files
+        self.tempdir = TemporaryDirectory()
+        self.train_file = os.path.join(self.tempdir.name, "train.dat")
+        self.test_file = os.path.join(self.tempdir.name, "test.dat")
+        self.model_file = os.path.join(self.tempdir.name, "vw.model")
+        self.prediction_file = os.path.join(self.tempdir.name, "prediction.dat")
+
+        # set DataFrame columns
+        self.col_user = col_user
+        self.col_item = col_item
+        self.col_rating = col_rating
+        self.col_timestamp = col_timestamp
+        self.col_prediction = col_prediction
+
+        self.logistic = "logistic" in kwargs.values()
+        self.train_cmd = self.parse_train_params(params=kwargs)
+        self.test_cmd = self.parse_test_params(params=kwargs)
+
+    @staticmethod
```

Comment:
why static?

Reply:
just because it doesn't access any class members